### PR TITLE
PostgreSQL - SQL error when search for module

### LINKED ISSUE #6159: Postgresql: Search in Module Manager gives an error
Go to the Module Manager in Backend and try to search for a term. You should get this error:

Error

SQL=SELECT a.id, a.title, a.note, a.position, a.module, a.language,a.checked_out, a.checked_out_time, a.published+2*(e.enabled-1) as published, a.access, a.ordering, a.publish_up, a.publish_down,l.title AS language_title,uc.name AS editor,ag.title AS access_level,MIN(mm.menuid) AS pages,e.name AS name FROM "invtf_modules" AS a LEFT JOIN "invtf_languages" AS l ON l.lang_code = a.language LEFT JOIN invtf_users AS uc ON uc.id=a.checked_out LEFT JOIN invtf_viewlevels AS ag ON ag.id = a.access LEFT JOIN invtf_modules_menu AS mm ON mm.moduleid = a.id LEFT JOIN invtf_extensions AS e ON e.element = a.module WHERE (a.published IN (0, 1)) AND a.client_id = 0 AND e.client_id =0 AND (a.title REGEXP 'tag' OR a.note REGEXP 'tag') GROUP BY a.id, a.title, a.note, a.position, a.module, a.language,a.checked_out,a.checked_out_time, a.published, a.access, a.ordering,l.title, uc.name, ag.title, e.name,l.lang_code, uc.id, ag.id, mm.moduleid, e.element, a.publish_up, a.publish_down,e.enabled SQL=SELECT a.id, a.title, a.note, a.position, a.module, a.language,a.checked_out, a.checked_out_time, a.published+2*(e.enabled-1) as published, a.access, a.ordering, a.publish_up, a.publish_down,l.title AS language_title,uc.name AS editor,ag.title AS access_level,MIN(mm.menuid) AS pages,e.name AS name FROM "invtf_modules" AS a LEFT JOIN "invtf_languages" AS l ON l.lang_code = a.language LEFT JOIN invtf_users AS uc ON uc.id=a.checked_out LEFT JOIN invtf_viewlevels AS ag ON ag.id = a.access LEFT JOIN invtf_modules_menu AS mm ON mm.moduleid = a.id LEFT JOIN invtf_extensions AS e ON e.element = a.module WHERE (a.published IN (0, 1)) AND a.client_id = 0 AND e.client_id =0 AND (a.title REGEXP 'tag' OR a.note REGEXP 'tag') GROUP BY a.id, a.title, a.note, a.position, a.module, a.language,a.checked_out,a.checked_out_time, a.published, a.access, a.ordering,l.title, uc.name, ag.title, e.name,l.lang_code, uc.id, ag.id, mm.moduleid, e.element, a.publish_up, a.publish_down,e.enabled ORDER BY "position" asc,a.ordering ASC LIMIT 20 SQL=SELECT a.id, a.title, a.note, a.position, a.module, a.language,a.checked_out, a.checked_out_time, a.published+2*(e.enabled-1) as published, a.access, a.ordering, a.publish_up, a.publish_down,l.title AS language_title,uc.name AS editor,ag.title AS access_level,MIN(mm.menuid) AS pages,e.name AS name FROM "invtf_modules" AS a LEFT JOIN "invtf_languages" AS l ON l.lang_code = a.language LEFT JOIN invtf_users AS uc ON uc.id=a.checked_out LEFT JOIN invtf_viewlevels AS ag ON ag.id = a.access LEFT JOIN invtf_modules_menu AS mm ON mm.moduleid = a.id LEFT JOIN invtf_extensions AS e ON e.element = a.module WHERE (a.published IN (0, 1)) AND a.client_id = 0 AND e.client_id =0 AND (a.title REGEXP 'tag' OR a.note REGEXP 'tag') GROUP BY a.id, a.title, a.note, a.position, a.module, a.language,a.checked_out,a.checked_out_time, a.published, a.access, a.ordering,l.title, uc.name, ag.title, e.name,l.lang_code, uc.id, ag.id, mm.moduleid, e.element, a.publish_up, a.publish_down,e.enabled ORDER BY "position" asc,a.ordering ASC

### PR DESCRIPTION
**Pull request by alikon**
#### Steps to reproduce the issue

Go to the Extension -> Module Manager in Backend  and  search for a term
#### Expected result

If some match are found the result module list is returned
#### Actual result

Got an ERROR  SQL QUERY SYNTAX ERROR  REGEXP
#### System information

PostgreSQL 9.3.5
Joomla 3.4.0
#### Additional comments

the `REGEXP` operator work only for Mysql  
for Postgres we should use  `~*`

fix #6159

### CODE REVIEW

**Comment by mbabker** [comment]
It looks like this is going to fail as you added the method to the query classes yet you're calling it from the driver class.

Also, we need a method in the base driver/query class (whichever it is supposed to be in) otherwise it'll need to be wrapped in a `method_exists()` check as I don't see a method here for SQL Server.

**Comment by alikon** [comment]
@mbabker you are right my fault,  hope better now

for the SQL server  based on this http://www.sql-workbench.net/dbms_comparison.html we are in trouble

so i ask  why we need a posix pattern matching in module/template/style search ?
for db's portability should be better the trivial LIKE

**Comment by mbabker** [comment]
Can't say I have an answer for that one, but if it works the same with a LIKE then go for it.

**Comment by alikon** [comment]
i've switched the 3 models from REGEXP to LIKE

despite i don't have a clue for MSSQL  posix pattern matching i've added the "portability" method regexp for Mysql and Postgresql  for future use

File: libraries/joomla/database/query/mysqli.php [hunk 1]
```diff
@@ -101,4 +101,23 @@ public function setLimit($limit = 0, $offset = 0)
...
+	 * Usage:
+	 * $query->where('field ' . $db->regexp($search));
+	 *
+	 * @param   string  $value  The regex pattern.
+	 *
+	 * @return  string  Returns the regex operator.
+	 *
+	 * @since   11.3
+	 */
+	public function regexp($value)
```
Comment (gunjanpatel) [inline libraries/joomla/database/query/mysqli.php]: Good work @alikon 
You have not used this functions anywhere. So, this doesn't need anymore or it's for future purpose?

Comment (gunjanpatel) [inline libraries/joomla/database/query/mysqli.php]: and I think you should add function here `https://github.com/joomla/joomla-cms/blob/staging/libraries/joomla/database/driver/postgresql.php` if you want to use it like, `$db->regex()` in query. Similarly, `$db->escape()`

**Comment by alikon** [comment]
It's for future purpose 
The Regexp is an operator and usually it's used mainly in  where condition or this is the way I think at
So I've putted  that method in the query class

**Comment by alikon** [comment]
@gunjanpatel 
as usual my typos in the comment  i've writed
```
* Usage:
* $query->where('field ' . $db->regexp($search));
```
instead of
```
$query->where('field ' . $query->regexp($search));
```

**Comment by gunjanpatel** [comment]
Super @alikon Thanks.

**Comment by ketchupmonki** [comment]
Successfully tested in postgre.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6188">issues.joomla.org/joomla-cms/6188</a>.</sub>

**Comment by waader** [comment]
@test works with postgresql, mssql, mysql! Thanks @alikon!

It also fixes search in template manager for postgresql and mssql.

**Comment by zero-24** [comment]
Moving to RTC! Thanks for coding and testing!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6188">issues.joomla.org/joomla-cms/6188</a>.</sub>

**Comment by Kubik-Rubik** [comment]
Great work @alikon!

**Pull request merged**